### ISSUE TITLE
IndexOutOfBoundsException in version 1.1.3

### ISSUE THREAD

**Issue by jahunt1** [intercom/intercom-android#149]
```
Fatal Exception: java.lang.IndexOutOfBoundsException: Invalid index 2, size is 2
       at java.util.ArrayList.throwIndexOutOfBoundsException(ArrayList.java:255)
       at java.util.ArrayList.get(ArrayList.java:308)
       at android.widget.ArrayAdapter.getItem(ArrayAdapter.java:337)
       at io.intercom.android.sdk.conversation.ConversationAdapter.isEnabled(ConversationAdapter.java:142)
       at android.widget.ListView.dispatchDraw(ListView.java:3387)
       at android.view.View.draw(View.java:16475)
       at android.widget.AbsListView.draw(AbsListView.java:6056)
       at android.view.View.updateDisplayListIfDirty(View.java:15402)
       at android.view.View.getDisplayList(View.java:15424)
       at android.view.View.draw(View.java:16194)
       at android.view.ViewGroup.drawChild(ViewGroup.java:3709)
       at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3502)
       at android.view.View.draw(View.java:16475)
       at android.view.View.updateDisplayListIfDirty(View.java:15402)
       at android.view.View.getDisplayList(View.java:15424)
       at android.view.View.draw(View.java:16194)
       at android.view.ViewGroup.drawChild(ViewGroup.java:3709)
       at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3502)
       at android.view.View.updateDisplayListIfDirty(View.java:15397)
       at android.view.View.getDisplayList(View.java:15424)
       at android.view.View.draw(View.java:16194)
       at android.view.ViewGroup.drawChild(ViewGroup.java:3709)
       at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3502)
       at android.view.View.draw(View.java:16475)
       at android.view.View.updateDisplayListIfDirty(View.java:15402)
       at android.view.View.getDisplayList(View.java:15424)
       at android.view.View.draw(View.java:16194)
       at android.view.ViewGroup.drawChild(ViewGroup.java:3709)
       at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3502)
       at android.view.View.updateDisplayListIfDirty(View.java:15397)
       at android.view.View.getDisplayList(View.java:15424)
       at android.view.View.draw(View.java:16194)
       at android.view.ViewGroup.drawChild(ViewGroup.java:3709)
       at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3502)
       at android.view.View.updateDisplayListIfDirty(View.java:15397)
       at android.view.View.getDisplayList(View.java:15424)
       at android.view.View.draw(View.java:16194)
       at android.view.ViewGroup.drawChild(ViewGroup.java:3709)
       at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3502)
       at android.view.View.draw(View.java:16475)
       at android.widget.FrameLayout.draw(FrameLayout.java:598)
       at com.android.internal.policy.impl.PhoneWindow$DecorView.draw(PhoneWindow.java:3107)
       at android.view.View.updateDisplayListIfDirty(View.java:15402)
       at android.view.View.getDisplayList(View.java:15424)
       at android.view.ThreadedRenderer.updateViewTreeDisplayList(ThreadedRenderer.java:310)
       at android.view.ThreadedRenderer.updateRootDisplayList(ThreadedRenderer.java:316)
       at android.view.ThreadedRenderer.draw(ThreadedRenderer.java:355)
       at android.view.ViewRootImpl.draw(ViewRootImpl.java:2889)
       at android.view.ViewRootImpl.performDraw(ViewRootImpl.java:2683)
       at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2270)
       at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1264)
       at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:6944)
       at android.view.Choreographer$CallbackRecord.run(Choreographer.java:777)
       at android.view.Choreographer.doCallbacks(Choreographer.java:590)
       at android.view.Choreographer.doFrame(Choreographer.java:560)
       at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:763)
       at android.os.Handler.handleCallback(Handler.java:739)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:145)
       at android.app.ActivityThread.main(ActivityThread.java:6837)
       at java.lang.reflect.Method.invoke(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:372)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1404)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1199)
```

The problem appears to be in the ```ConversationAdapter```

**Comment by harkin** [comment]
Thanks for reporting this one. I'll look into what's happening here. How often are you seeing this?

**Comment by jahunt1** [comment]
This is the only time I have seen this particular crash. The crash report indicated this happened on a SM-G920V running Android 5.1.1. In the meantime, we have updated our build config to use a newer version of the Intercom SDK.

**Comment by DaleCantwell** [comment]
@jahunt1 I'm going to close this issue out for now as I haven't seen any reports of it on our newest SDK version. Ping us if you encounter this on one of our newer version.

**Comment by bsiegel** [comment]
We've experienced this in v1.1.9 of the SDK. We will be updating to use the most recent version, but I don't see anything in the changelogs to indicate this issue might have been fixed between 1.1.9 and 1.1.11.

Device:	SM-N915R4, OS Version:	Android 5.0.1
Device:	LGUS991, OS Version:	Android 5.1

```
java.lang.IndexOutOfBoundsException: Invalid index 10, size is 10
       at java.util.ArrayList.throwIndexOutOfBoundsException(ArrayList.java:255)
       at java.util.ArrayList.get(ArrayList.java:308)
       at android.widget.ArrayAdapter.getItem(ArrayAdapter.java:337)
       at io.intercom.android.sdk.conversation.ConversationAdapter.isEnabled(ConversationAdapter.java:142)
       at android.widget.ListView.dispatchDraw(ListView.java:3307)
       at android.view.View.draw(View.java:15311)
       at android.widget.AbsListView.draw(AbsListView.java:4345)
       at android.view.View.updateDisplayListIfDirty(View.java:14235)
       at android.view.View.getDisplayList(View.java:14257)
       at android.view.View.draw(View.java:15036)
       at android.view.ViewGroup.drawChild(ViewGroup.java:3408)
       at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3201)
       at android.view.View.draw(View.java:15311)
       at android.view.View.updateDisplayListIfDirty(View.java:14235)
       at android.view.View.getDisplayList(View.java:14257)
       at android.view.View.draw(View.java:15036)
       at android.view.ViewGroup.drawChild(ViewGroup.java:3408)
       at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3201)
       at android.view.View.updateDisplayListIfDirty(View.java:14230)
       at android.view.View.getDisplayList(View.java:14257)
       at android.view.View.draw(View.java:15036)
       at android.view.ViewGroup.drawChild(ViewGroup.java:3408)
       at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3201)
       at android.view.View.draw(View.java:15311)
       at android.view.View.updateDisplayListIfDirty(View.java:14235)
       at android.view.View.getDisplayList(View.java:14257)
       at android.view.View.draw(View.java:15036)
       at android.view.ViewGroup.drawChild(ViewGroup.java:3408)
       at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3201)
       at android.view.View.updateDisplayListIfDirty(View.java:14230)
       at android.view.View.getDisplayList(View.java:14257)
       at android.view.View.draw(View.java:15036)
       at android.view.ViewGroup.drawChild(ViewGroup.java:3408)
       at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3201)
       at android.view.View.updateDisplayListIfDirty(View.java:14230)
       at android.view.View.getDisplayList(View.java:14257)
       at android.view.View.draw(View.java:15036)
       at android.view.ViewGroup.drawChild(ViewGroup.java:3408)
       at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3201)
       at android.view.View.draw(View.java:15311)
       at android.widget.FrameLayout.draw(FrameLayout.java:598)
       at com.android.internal.policy.impl.PhoneWindow$DecorView.draw(PhoneWindow.java:2748)
       at android.view.View.updateDisplayListIfDirty(View.java:14235)
       at android.view.View.getDisplayList(View.java:14257)
       at android.view.ThreadedRenderer.updateViewTreeDisplayList(ThreadedRenderer.java:278)
       at android.view.ThreadedRenderer.updateRootDisplayList(ThreadedRenderer.java:284)
       at android.view.ThreadedRenderer.draw(ThreadedRenderer.java:323)
       at android.view.ViewRootImpl.draw(ViewRootImpl.java:2579)
       at android.view.ViewRootImpl.performDraw(ViewRootImpl.java:2393)
       at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2023)
       at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1085)
       at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:5991)
       at android.view.Choreographer$CallbackRecord.run(Choreographer.java:767)
       at android.view.Choreographer.doCallbacks(Choreographer.java:580)
       at android.view.Choreographer.doFrame(Choreographer.java:550)
       at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:753)
       at android.os.Handler.handleCallback(Handler.java:739)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:135)
       at android.app.ActivityThread.main(ActivityThread.java:5431)
       at java.lang.reflect.Method.invoke(Native Method)
       at java.lang.reflect.Method.invoke(Method.java:372)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:913)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:706)
```

**Comment by trevjonez** [comment]
Smells a lot like the issue resolved here with a questionable try catch: 
http://stackoverflow.com/questions/8431342/listview-random-indexoutofboundsexception-on-froyo
